src/Subscriptions/Iterators/GuardContextSyncIterator.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Nuwave\Lighthouse\Subscriptions\Iterators;
+
+use Closure;
+use Exception;
+use Illuminate\Contracts\Auth\Factory;
+use Illuminate\Contracts\Config\Repository;
+use Illuminate\Support\Collection;
+use Nuwave\Lighthouse\Subscriptions\Contracts\SubscriptionIterator;
+use Nuwave\Lighthouse\Subscriptions\Subscriber;
+use Nuwave\Lighthouse\Subscriptions\SubscriptionGuard;
+
+class GuardContextSyncIterator implements SubscriptionIterator
```

Comment:
I would like a more descriptive name for that. How about `LoginSubscriberIterator`?

Reply:
I had a hard timing thinking of a good name. How about `AuthenticatedSubscriberIterator`?

Reply:
`AuthenticatingSubscriberIterator`?

Reply:
I will change it.

Reply:
Sounds good to me!